network/src/network_service.rs

```diff
@@ -125,4 +126,14 @@ impl NetworkService {
     pub fn add_node(&self, peer_id: &PeerId, address: Multiaddr) {
         self.network.add_node(peer_id, address)
     }
+
+    pub fn connected_peers(&self) -> Vec<(PeerId, Peer)> {
+        // TODO return listen addr for connected peers
+        self.network
+            .peers_registry
+            .read()
+            .peers_iter()
+            .map(|(peer_id, peer)| (peer_id.clone(), peer.clone()))
+            .collect()
```

Comment:
```
let peer_store = self.network.peer_store.read();
let (peer_id, peer, peer_addresses) = peers.map(|(peer_id, peer)| (peer_id.clone(), peer.clone(), peer_store.peer_addrs(peer_id, 3))).collect();
```

Reply:
`peer_addrs` still returns incorrect port:

```json
{
  "jsonrpc": "2.0",
  "result": [
    {
      "addresses": [
        {
          "address": "/ip4/0.0.0.0/tcp/8117",
          "score": 1
        },
        {
          "address": "/ip4/127.0.0.1/tcp/55676/p2p/QmdiJuQZj1dM4K4HKMxfMwcAqGFYvGKpbvVxTzyQeNGEcG",
          "score": 1
        }
      ],
      "node_id": "QmdiJuQZj1dM4K4HKMxfMwcAqGFYvGKpbvVxTzyQeNGEcG",
      "version": "unknown"
    }
  ],
  "id": 2
}
```

I will leave this issue here, please help to add more assert in integration test after #351 is merged.